src/Client.cpp

```diff
@@ -153,50 +177,62 @@ int UrsaClient::start() {
         }
     }
 
-    // wait for status thread to terminate
-    status_th.join();
+    status_th.detach();
 
     return 0;
 }
 
-UrsaClient::UrsaClient(std::string server_addr, std::string db_command)
-    : server_addr(server_addr), db_command(db_command) {}
+UrsaClient::UrsaClient(std::string server_addr, std::string db_command,
+                       bool silent_mode)
+    : server_addr(server_addr),
+      db_command(db_command),
+      silent_mode(silent_mode) {}
 
 static void print_usage(const char *arg0) {
-    spdlog::info("Usage: {} [server_addr] [db_command]", arg0);
+    spdlog::info("Usage: {} [server_addr] [args...]", arg0);
     spdlog::info(
-        "    server_addr - server connection string, default: "
+        "    [server_addr]      server connection string, default: "
         "tcp://localhost:9281");
     spdlog::info(
-        "    db_command - specific command to be run in the database, "
+        "    [-q <db_command>]  specific command to be run in the database, "
         "if not provided - interactive mode");
+    spdlog::info(
+        "    [-s]               silent mode, dump only command output");
 }
 
-int main(int argc, const char *argv[]) {
+int main(int argc, char *argv[]) {
     std::string server_addr = "tcp://localhost:9281";
     std::string db_command = "";
-
-    if (argc >= 2) {
-        std::string first_arg = std::string(argv[1]);
-
-        if (first_arg == "-h" || first_arg == "--help") {
-            print_usage(argc >= 1 ? argv[0] : "ursacli");
-            return 0;
+    bool silent_mode = false;
+
+    int c;
+
+    while ((c = getopt(argc, argv, "hsq:")) != -1) {
+        switch (c) {
+            case 's':
+                silent_mode = true;
+                break;
+            case 'q':
+                db_command = optarg;
+                break;
+            case 'h':
+                print_usage(argc >= 1 ? argv[0] : "ursacli");
+                return 0;
+            default:
+                print_usage(argc >= 1 ? argv[0] : "ursacli");
```

Comment:
I think we can assume that argc is >= 1. If someone does cursed things, they're on their own.

Reply:
What if someone starts ursacli from their shellcode using execve with an empty argv array?

Reply:
:duck:

Reply:
idk, this doesn't cost anything bug we have strong assertion of not going out of bounds here